bugshaker/src/main/java/com/github/stkent/bugshaker/MainActivity.java

```diff
@@ -0,0 +1,279 @@
+		largeEraser = getResources().getInteger(R.integer.large_eraser_size);
+
+		ImageButton drawBtn = (ImageButton) findViewById(R.id.draw_btn);
+		drawBtn.setOnClickListener(this);
+
+		drawView.setBrushSize(smallBrush);
+
+		ImageButton eraseBtn = (ImageButton) findViewById(R.id.erase_btn);
+		eraseBtn.setOnClickListener(this);
+
+		ImageButton sendBtn = (ImageButton) findViewById(R.id.sendEmail);
+		sendBtn.setOnClickListener(this);
+
+		ImageButton speechBtn = (ImageButton) findViewById(R.id.speechBox);
+		speechBtn.setOnClickListener(this);
+
+		final GenericEmailIntentProvider genericEmailIntentProvider
+			= new GenericEmailIntentProvider();
+
+		Logger logger = new Logger(true);
+		Application application = getApplication();
+
+		EmailCapabilitiesProvider emailCapabilitiesProvider = new EmailCapabilitiesProvider(
+			application.getPackageManager(), genericEmailIntentProvider, logger);
+
+		feedbackEmailFlowManager = new FeedbackEmailFlowManager(
+			application,
+			emailCapabilitiesProvider,
+			new Toaster(application),
+			new ActivityReferenceManager(),
+			new FeedbackEmailIntentProvider(application, genericEmailIntentProvider));
+
+		String pathOfScreenshot = getIntent().getStringExtra("uri");
+		File screenshotFile = new File(pathOfScreenshot);
+		if(screenshotFile.exists()) {
+			Bitmap myBitmap = BitmapFactory.decodeFile(pathOfScreenshot);
+
+			drawView = (DrawingView) findViewById(R.id.drawing);
+			Drawable temp = new BitmapDrawable(getResources(), myBitmap);
+			drawView.setBackgroundDrawable(temp);
+		}
+		else {
+			throw new RuntimeException();
+		}
+	}
+
+	@Override
+	public void onClick(View view){
+		//respond to clicks
+		if(view.getId()==R.id.draw_btn){
+			//draw button clicked
+			final Dialog brushDialog = new Dialog(this);
+			brushDialog.setTitle(getApplicationContext().getString(R.string.brush_dialog_title));
+			brushDialog.setContentView(R.layout.brush_chooser);
+
+			ImageButton smallBtn = (ImageButton) brushDialog.findViewById(R.id.small_brush);
+			smallBtn.setOnClickListener(new OnClickListener() {
+				@Override
+				public void onClick(View view) {
+					drawView.setBrushSize(smallBrush);
+					drawView.setLastBrushSize(smallBrush);
+					drawView.setErase(false);
+					brushDialog.dismiss();
+				}
+			});
+
+			ImageButton mediumBtn = (ImageButton) brushDialog.findViewById(R.id.medium_brush);
+			mediumBtn.setOnClickListener(new OnClickListener() {
+				@Override
+				public void onClick(View view) {
+					drawView.setBrushSize(mediumBrush);
+					drawView.setLastBrushSize(mediumBrush);
+					drawView.setErase(false);
+					brushDialog.dismiss();
+				}
+			});
+
+			ImageButton largeBtn = (ImageButton) brushDialog.findViewById(R.id.large_brush);
+			largeBtn.setOnClickListener(new OnClickListener() {
+				@Override
+				public void onClick(View view) {
+					drawView.setBrushSize(largeBrush);
+					drawView.setLastBrushSize(largeBrush);
+					drawView.setErase(false);
+					brushDialog.dismiss();
```

Comment:
the code for clicklisteners on the brush buttons is really identical, except for the brushSize. Can we extract this out to a method here?(We don't need a util for it though)